pkg/config/config.go

```diff
@@ -185,6 +186,26 @@ func (c *Config) Validate(fs *flag.FlagSet) error {
 
 	c.Metrics.ServiceConfig.APIEnableGetConfiguration = c.EnableConfigEndpoints
 
+	// find every traces instance that is to be used by app o11y receiver
+	// and configure it to include a PushReceiver to be able to push traces internally
+	// this cannot happen in unmarshall stage since integrations are not yet unmarshalled
+	if c.Integrations.configV2 != nil {
+		for _, conf := range c.Integrations.configV2.Configs {
+			if conf.Name() != app_o11y_receiver.IntegrationName {
+				continue
+			}
+			o11yConf := conf.(*app_o11y_receiver.Config)
+			if o11yConf.ReceiverConfig.TracesInstance == "" {
+				continue
+			}
+			for idx, instanceConf := range c.Traces.Configs {
+				if instanceConf.Name == o11yConf.ReceiverConfig.TracesInstance {
+					c.Traces.Configs[idx].PushReceiver = true
+				}
+			}
+		}
+	}
```

Comment:
Can we have every trace instance have an internal push receiver by default? That we don't need a special loop here.

Reply:
That was original impl, but @mapno  suggested to add only when necessary. Shall I revert to including by default?